tests/test_evesettings_names.py

```
@@ -0,0 +1,214 @@
+    # Second pass has nothing missing, so nothing was learned.
+    assert cache.resolve_missing([1], fetch=fetch) is False
+
+
+def test_cache_labels_unresolved_ids_with_a_fallback():
+    cache = names.NameCache()
+    assert cache.label(98123456) == "Character 98123456"
+    cache.names[98123456] = "Pilot"
+    assert cache.label(98123456) == "Pilot"
+
+
+def test_fetch_batch_classifies_an_http_error(monkeypatch):
```

Ruff (0.16.2)

[warning] Unused function argument: `monkeypatch`

(ARG001)